lib/symboldatabase.cpp

```diff
@@ -7185,7 +7185,7 @@ static const Token* parsedecl(const Token* type,
             else if (enum_type->isUnsigned())
                 valuetype->sign = ValueType::Sign::UNSIGNED;
             else
-                valuetype->sign = defaultSignedness;
+                valuetype->sign = defaultSignedness; // TODO: this is implementation-dependent might be separate from char
```

Comment:
What do you mean? I feel there is something I don't understand.
```
enum {A,B,C} abc;
if (abc == A) {}  // <- A is an 'int' right?
```

Reply:
It is an integer but not necessarily `int`.

From https://en.cppreference.com/w/cpp/language/enum:
`the underlying type is an implementation-defined integral type that can represent all enumerator values; this type is not larger than int unless the value of an enumerator cannot fit in an int or unsigned int`

Also https://stackoverflow.com/a/1122128/532627 for an actual reference to the standard.

It should definitely not depend on the flag we use for `char` which is configurable via compiler flags (and the CLI).

Reply:
> It is an integer but not necessarily int.

ok I see.

> It should definitely not depend on the flag we use for char which is configurable via compiler flags.

I agree.

Reply:
I filed https://trac.cppcheck.net/ticket/13602 about this.